### ISSUE TITLE
Dependency generation issued by the wmake stack is generating duplicate calls

### ISSUE THREAD

**Issue by wyldckat** [blueCFD/Core#69]
This is fairly complicated to fix, because this was in itself a consequence of a previous fix that was done in blueCFD-Core 2016-2 - [see the release notes, entry 25](http://bluecfd.github.io/Core/ReleaseNotes/bluecfd-core-2016-2/#cumulative-changes-and-notes).

For example, building the library `src/mesh/snappyHexMesh` will give the following list of dependencies:

```
Making dependency list for source file zeroFixedValuePointPatchFields.C
Making dependency list for source file medialAxisMeshMover.C
Making dependency list for source file externalDisplacementMeshMover.C
Making dependency list for source file displacementMeshMoverMotionSolver.C
Making dependency list for source file externalDisplacementMeshMover/zeroFixedValue/zeroFixedValuePointPatchFields.C
Making dependency list for source file trackedParticleCloud.C
Making dependency list for source file externalDisplacementMeshMover/externalDisplacementMeshMover.C
Making dependency list for source file trackedParticle.C
Making dependency list for source file externalDisplacementMeshMover/displacementMeshMoverMotionSolver.C
Making dependency list for source file shellSurfaces.C
Making dependency list for source file shellSurfaces/shellSurfaces.C
Making dependency list for source file trackedParticle/trackedParticleCloud.C
Making dependency list for source file trackedParticle/trackedParticle.C
Making dependency list for source file refinementSurfaces.C
Making dependency list for source file surfaceZonesInfo.C
Making dependency list for source file refinementFeatures.C
Making dependency list for source file refinementSurfaces/surfaceZonesInfo.C
Making dependency list for source file patchFaceOrientation.C
Making dependency list for source file externalDisplacementMeshMover/medialAxisMeshMover.C
Making dependency list for source file meshRefinementRefine.C
Making dependency list for source file refinementSurfaces/refinementSurfaces.C
Making dependency list for source file meshRefinementProblemCells.C
Making dependency list for source file meshRefinement/patchFaceOrientation.C
Making dependency list for source file meshRefinementMerge.C
Making dependency list for source file refinementFeatures/refinementFeatures.C
Making dependency list for source file meshRefinement.C
Making dependency list for source file meshRefinement/meshRefinementMerge.C
Making dependency list for source file meshRefinement/meshRefinementProblemCells.C
Making dependency list for source file meshRefinementBaffles.C
Making dependency list for source file pointData.C
Making dependency list for source file meshRefinement/meshRefinementRefine.C
Making dependency list for source file snappyHexMeshDriver/pointData/pointData.C
Making dependency list for source file snapParameters.C
Making dependency list for source file refinementParameters.C
Making dependency list for source file snappyHexMeshDriver/snapParameters/snapParameters.C
Making dependency list for source file layerParameters.C
Making dependency list for source file meshRefinement/meshRefinementBaffles.C
Making dependency list for source file snappyHexMeshDriver/refinementParameters/refinementParameters.C
Making dependency list for source file snappyRefineDriver.C
Making dependency list for source file snappySnapDriverFeature.C
Making dependency list for source file meshRefinement/meshRefinement.C
Making dependency list for source file snappySnapDriver.C
Making dependency list for source file snappyHexMeshDriver/layerParameters/layerParameters.C
Making dependency list for source file snappyHexMeshDriver/snappyRefineDriver.C
Making dependency list for source file snappyLayerDriver.C
Making dependency list for source file snappyHexMeshDriver/snappySnapDriverFeature.C
Making dependency list for source file snappyHexMeshDriver/snappySnapDriver.C
Making dependency list for source file snappyHexMeshDriver/snappyLayerDriver.C
```

Where there is a short and a long path for the same file, but the short path is more often than not a dud. This is due to the workaround we implemented for generating objects at the base folder of the object path, in this case at:

```
/home/ofuser/blueCFD/OpenFOAM-5.x/platforms/mingw_w64GccDPInt32Opt/src/mesh/snappyHexMesh/
```

Right now, the only solution I can think of requires adding to `as.exe` and `ld.exe` in the `binutils` package the support for UNC paths on Windows, as already done in `gcc` and `g++`.

**Comment by wyldckat** [comment]
Will try to implement and intermediate solution, along with fixing #70, namely to only use the long file name workaround when it is strictly needed.

**Comment by wyldckat** [comment]
OK, so the first reason as to why the dependency files were seemingly duplicate, because the script `wmake/rules/General/transform` already issues the message that `wmkdep` used to do.

Have fixed this in commit [a6a7e4f828936e7a7c387705cc0eb616f5473daf](https://github.com/blueCFD/OpenFOAM-dev/commit/a6a7e4f828936e7a7c387705cc0eb616f5473daf).

**Comment by wyldckat** [comment]
Once the change was made to `wmkdepend` and the adaptive LFN option is sent via `wmake`, the dependency generation list gave this:

```
Making dependency list for source file zeroFixedValuePointPatchFields.C
Making dependency list for source file medialAxisMeshMover.C
Making dependency list for source file externalDisplacementMeshMover.C
Making dependency list for source file displacementMeshMoverMotionSolver.C
Making dependency list for source file trackedParticleCloud.C
Making dependency list for source file trackedParticle.C
Making dependency list for source file shellSurfaces.C
Making dependency list for source file refinementSurfaces.C
Making dependency list for source file surfaceZonesInfo.C
Making dependency list for source file refinementFeatures.C
Making dependency list for source file patchFaceOrientation.C
Making dependency list for source file meshRefinementRefine.C
Making dependency list for source file meshRefinementProblemCells.C
Making dependency list for source file meshRefinementMerge.C
Making dependency list for source file meshRefinement.C
Making dependency list for source file meshRefinementBaffles.C
Making dependency list for source file pointData.C
Making dependency list for source file snapParameters.C
Making dependency list for source file refinementParameters.C
Making dependency list for source file layerParameters.C
Making dependency list for source file snappyRefineDriver.C
Making dependency list for source file snappySnapDriverFeature.C
Making dependency list for source file snappySnapDriver.C
Making dependency list for source file snappyLayerDriver.C
```
